### PR TITLE
MM-41850_Webapp: Update to New Channel Modal

### PR DESCRIPTION
**Pull request by julmondragon**
#### Summary
Update to New Channel Modal
- First stage creating and adjusting components required. **ONLY FOR INITIAL REVIEW**

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41850

#### Screenshots
**Input**
input

**URL Input**
url-input

**Public Private selector**
public-private

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/widgets/inputs/input/input.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (julmondragon) [inline components/widgets/inputs/input/input.tsx:1]: Moving https://github.com/mattermost/mattermost-webapp/blob/master/components/input.tsx to widgets and applying some enhancements required.
TODO: Migrate to func component.

**Comment by julmondragon** [comment]
For components in widgets the idea is to add jest unit testing and e2e testing using the storybooks.

File: components/widgets/public-private-selector/public-private-selector.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (julmondragon) [inline components/widgets/public-private-selector/public-private-selector.tsx:1]: Based on https://github.com/mattermost/mattermost-plugin-playbooks/blob/master/webapp/src/components/backstage/public_private_selector.tsx from Playbooks, handling use cases for both projects.

File: components/widgets/inputs/input/input.tsx (hunk 1 above)
Comment (marianunez) [reply components/widgets/inputs/input/input.tsx:1]: So this new component is based on https://github.com/mattermost/mattermost-webapp/blob/master/components/input.tsx? Once complete, will we remove the original one and update the other components to use this new one?

Comment (julmondragon) [reply components/widgets/inputs/input/input.tsx:1]: Correct, that's the idea.

**Comment by marianunez** [comment]
Swapping out reviewers as Julien will be OOO this week.

File: components/widgets/inputs/input/input.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (neallred) [inline components/widgets/inputs/input/input.stories.tsx:1]: I like the use of stories! They are perfect for a component like this.

**Pull request closed without merging**